legacy/src/main/java/org/opensearch/sql/legacy/executor/join/ElasticJoinExecutor.java

```diff
@@ -87,7 +93,23 @@ public void sendResponse(RestChannel channel) throws IOException {
   public void run() throws IOException, SqlParseException {
     try {
       long timeBefore = System.currentTimeMillis();
-      pit = new PointInTimeHandlerImpl(client, indices);
+
+      LOG.info("🔍 Starting join execution, checking for JOIN_TIME_OUT hints...");
+
+      // ✅ Extract JOIN_TIME_OUT hint and create PIT with custom keepalive
+      TimeValue customKeepAlive = extractJoinTimeoutFromHints();
+
+      if (customKeepAlive != null) {
```

Comment:
thought: since `run` is already a long function, we should probably extract the PIT construction to its own method for clarity.